test/unsafe_methods_test.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "minitest/autorun"
+require "rgeo/shapefile"
+
+class UnsafeMethodsTest < Minitest::Test
+  def test_unsafe_methods_true
+    _open_shapefile("test", true) do |file_|
+      rec_ = file_.next # Does not raise an error
+      refute_nil(rec_.geometry)
+      assert_equal(false, rec_.geometry.valid?)
+    end
+  end
+
+  def test_unsafe_methods_false
+    _open_shapefile("test", false) do |file_|
+      assert_raises(RGeo::Error::InvalidGeometry) { file_.next }
+    end
+  end
+
+  private
+
+  def _open_shapefile(name_, unsafe_methods_, &block_)
+    RGeo::Shapefile::Reader.open(
+      File.expand_path("ring_self_intersection/#{name_}",
+                       File.dirname(__FILE__)), unsafe_methods: unsafe_methods_,
```

Comment:
```suggestion
                       __dir__), unsafe_methods: unsafe_methods_,
```

This makes me think about how we are missing rubocop. And looking at the tests we're still on travis, there is definitely some work here!

Reply:
I can update this with what's in rgeo in a follow up PR if that would be preferable. Good call here!

Reply:
Once again, the more you contribute the more grateful I'll be !

Reply:
https://github.com/rgeo/rgeo-shapefile/pull/51
😄

Reply:
@BuonOmo #47 actuallyh migrates to GHA from Travis. I can remove the RGeo rc dependency on that and it should be good to merge